.github/workflows/testing.yml

```diff
@@ -0,0 +1,49 @@
+name: Testing workflow
+
+on: [push]
+
+jobs:
+  style:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: "Install clang-format for Style Format Checking"
+        run: |
+          sudo apt update
+          sudo apt install clang-format
+      - name: "Run clang-format for Style Format checking"
+        run: clang-format -i src/*.hpp src/*.cpp tests/*.cpp --dry-run --Werror
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: "Install necessary dependencies for C++"
+        run: sudo apt update && sudo apt install cmake gcc libeigen3-dev libyaml-cpp-dev libboost-all-dev
+      - name: "Create build folder"
+        run: mkdir build
+      - name: "Run cmake"
+        working-directory: ./build
+        run: cmake ..
+      - name: "Build make"
+        working-directory: ./build
+        run: make
+  test:
+    needs: build
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: "Install necessary dependencies for C++"
+        run: sudo apt update && sudo apt install cmake gcc libeigen3-dev libyaml-cpp-dev libboost-all-dev
+      - name: "Create build folder"
+        run: mkdir build
+      - name: "Run cmake"  
+        working-directory: ./build
+        run: cmake ..
+      - name: "Make"
+        working-directory: ./build
+        run: make -j4
```

Comment:
Here, you build again. For larger projects, this might not be what we want. We could instead reuse the build from the `build` job through an artifact.